### PR TITLE
Remove pudl@catalyst.coop from docs

### PR DESCRIPTION
**Pull request by bendnorman**
# PR Overview
#2665 removed references to pudl@catalyst.coop from datasette. This PR removes pudl@catalyst.coop from a couple of other public-facing pages to discourage folks from emailing pudl@catalyst.coop and instead asking questions on GitHub discussions.

Places `pudl@catalyst.coop` was not removed:

1. `pyproject.toml` and `src/pudl/__init__.py`: I think PyPi gets the email contact information from at least one of these files.
2. `CODE_OF_CONDUCT.md`: If folks need to report inappropriate behavior it should not be public!
3. `src/pudl/metadata/constants.py`, `src/pudl/metadata/sources.py`: I think the zenodo archives pull contact information from these files.

# PR Checklist

- [ ] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [ ] Make sure you've included good docstrings.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
- [ ] Include unit tests for new functions and classes.
- [ ] Defensive data quality/sanity checks in analyses & data processing functions.
- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [ ] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

**Review by e-belfer — APPROVED** [review]
These changes look good to me. Makes sense to leave the email in in the other places.

**Review by aesharpe — COMMENTED** [review]
These look good! 
I ran the docs build and found a few more that weren't mentioned in the comments: 
- The Data Access page still references pudl.catalyst.coop:

File: README.rst [hunk 1]
```diff
@@ -250,9 +250,9 @@ Contact Us
   `Office Hours <https://calend.ly/catalyst-cooperative/pudl-office-hours>`__
 * Follow us on Twitter: `@CatalystCoop <https://twitter.com/CatalystCoop>`__
 * More info on our website: https://catalyst.coop
-* For private communication about the project or to hire us to provide customized data
+* To hire us to provide customized data
   extraction and analysis, you can email the maintainers:
-  `pudl@catalyst.coop <mailto:pudl@catalyst.coop>`__
+  `hello@catalyst.coop <mailto:hello@catalyst.coop>`__
```
Comment (aesharpe) [inline README.rst:255]: The "Contact Us" section on the docs index page says something similar but uses the pudl@catalyst.coop email: 

> For private communication about the project or to hire us to provide customized data extraction and analysis, you can email the maintainers: [pudl@catalyst.coop](mailto:pudl%40catalyst.coop)

We should pick either hello@catalyst or pudl@catalyst and keep it the same for both.

**Review by aesharpe — CHANGES_REQUESTED** [review]

**Comment by bendnorman** [comment]
Hey @aesharpe! I rebuilt the docs locally and viewed the docs built by the RTD CI, and I couldn't find the errors you mentioned. Sometimes I need to force reload the docs built by the RTD CI to see the new changes.

**Comment by aesharpe** [comment]
> Hey @aesharpe! I rebuilt the docs locally and viewed the docs built by the RTD CI, and I couldn't find the errors you mentioned. Sometimes I need to force reload the docs built by the RTD CI to see the new changes.

Hmmm I wonder if I ran the docs build on the wrong branch or something. It also looks fine to me. In which case, smash merge!

**Review by aesharpe — APPROVED** [review]

**Pull request merged**